packages/core/src/extensions/Blocks/api/blocks/types.ts

```diff
@@ -77,130 +92,186 @@ export type TiptapBlockImplementation<T extends BlockConfig> = {
 
 // Container for both the config and implementation of a block,
 // and the type of BlockImplementation is based on that of the config
-export type BlockSpec<T extends BlockConfig> = {
+export type BlockSpec<
+  T extends BlockConfig,
+  B extends BlockSchema,
+  I extends InlineContentSchema,
+  S extends StyleSchema
+> = {
   config: T;
-  implementation: TiptapBlockImplementation<T>;
+  implementation: TiptapBlockImplementation<T, B, I, S>;
 };
 
 // Utility type. For a given object block schema, ensures that the key of each
 // block spec matches the name of the TipTap node in it.
-type NamesMatch<Blocks extends Record<string, BlockSpec<any>>> =
-  Blocks extends {
-    [Type in keyof Blocks]: Type extends string
-      ? Blocks[Type]["config"] extends { type: Type }
-        ? Blocks[Type]
-        : never
-      : never;
-  }
-    ? Blocks
+type NamesMatch<Blocks extends Record<string, BlockConfig>> = Blocks extends {
+  [Type in keyof Blocks]: Type extends string
+    ? Blocks[Type] extends { type: Type }
+      ? Blocks[Type]
+      : never
     : never;
+}
+  ? Blocks
+  : never;
 
 // Defines multiple block specs. Also ensures that the key of each block schema
 // is the same as name of the TipTap node in it. This should be passed in the
 // `blocks` option of the BlockNoteEditor. From a block schema, we can derive
 // both the blocks' internal implementation (as TipTap nodes) and the type
 // information for the external API.
-export type BlockSchema = NamesMatch<Record<string, BlockSpec<any>>>;
+export type BlockSchema = NamesMatch<Record<string, BlockConfig>>;
+
+export type BlockSpecs = Record<
+  string,
+  BlockSpec<any, any, InlineContentSchema, StyleSchema>
+>;
+
+export type BlockImplementations = Record<
+  string,
+  TiptapBlockImplementation<any, any, any, any>
+>;
+
+export type BlockSchemaFromSpecs<T extends BlockSpecs> = {
+  [K in keyof T]: T[K]["config"];
+};
 
 export type BlockSchemaWithBlock<
   BType extends string,
   C extends BlockConfig
 > = {
-  [k in BType]: BlockSpec<C>;
+  [k in BType]: C;
 };
 
-export type TableContent = {
+export type TableContent<
```

Comment:
hmm, I think not, because this is not a type of "inline content". it's a type of "block content", which for most blocks will be "inline content" / none, except for tables, where it is "table content"